src/bluehawk/processor/Processor.ts

```diff
@@ -23,16 +23,22 @@ export type Listener = (result: ParseResult) => void | Promise<void>;
 
 export class Processor {
   processors: CommandProcessors = {};
-  listeners: Listener[] = [];
+  listeners = new Set<Listener>();
 
   // Subscribe to processed file events
   subscribe(listener: Listener): void {
-    this.listeners.push(listener);
+    if (this.listeners.has(listener)) {
+      console.warn(`Skipping already registered listener: ${listener}`);
+      return;
+    }
+    this.listeners.add(listener);
   }
 
   // Publish a processed file
   async publish(result: ParseResult): Promise<void> {
```

Comment:
Hmm. Should the Processor halt and resume based on whether a listener succeeds in whatever task it is performing? It feels a bit counter to true pub/sub?